src/server/protocol.ts

```diff
@@ -3478,292 +3074,53 @@ export const enum IndentStyle {
-    allowJs?: boolean;
-    allowSyntheticDefaultImports?: boolean;
-    allowUnreachableCode?: boolean;
-    allowUnusedLabels?: boolean;
-    alwaysStrict?: boolean;
-    baseUrl?: string;
-    charset?: string;
-    checkJs?: boolean;
-    declaration?: boolean;
-    declarationDir?: string;
-    disableSizeLimit?: boolean;
-    downlevelIteration?: boolean;
-    emitBOM?: boolean;
-    emitDecoratorMetadata?: boolean;
-    experimentalDecorators?: boolean;
-    forceConsistentCasingInFileNames?: boolean;
-    importHelpers?: boolean;
-    inlineSourceMap?: boolean;
-    inlineSources?: boolean;
-    isolatedModules?: boolean;
-    jsx?: JsxEmit | ts.JsxEmit;
-    lib?: string[];
-    locale?: string;
-    mapRoot?: string;
-    maxNodeModuleJsDepth?: number;
-    module?: ModuleKind | ts.ModuleKind;
-    moduleResolution?: ModuleResolutionKind | ts.ModuleResolutionKind;
-    newLine?: NewLineKind | ts.NewLineKind;
-    noEmit?: boolean;
-    noEmitHelpers?: boolean;
-    noEmitOnError?: boolean;
-    noErrorTruncation?: boolean;
-    noFallthroughCasesInSwitch?: boolean;
-    noImplicitAny?: boolean;
-    noImplicitReturns?: boolean;
-    noImplicitThis?: boolean;
-    noUnusedLocals?: boolean;
-    noUnusedParameters?: boolean;
-    noImplicitUseStrict?: boolean;
-    noLib?: boolean;
-    noResolve?: boolean;
-    out?: string;
-    outDir?: string;
-    outFile?: string;
-    paths?: MapLike<string[]>;
-    plugins?: PluginImport[];
-    preserveConstEnums?: boolean;
-    preserveSymlinks?: boolean;
-    project?: string;
-    reactNamespace?: string;
-    removeComments?: boolean;
-    references?: ProjectReference[];
-    rootDir?: string;
-    rootDirs?: string[];
-    skipLibCheck?: boolean;
-    skipDefaultLibCheck?: boolean;
-    sourceMap?: boolean;
-    sourceRoot?: string;
-    strict?: boolean;
-    strictNullChecks?: boolean;
-    suppressExcessPropertyErrors?: boolean;
-    suppressImplicitAnyIndexErrors?: boolean;
-    useDefineForClassFields?: boolean;
-    target?: ScriptTarget | ts.ScriptTarget;
-    traceResolution?: boolean;
-    resolveJsonModule?: boolean;
-    types?: string[];
-    /** Paths used to used to compute primary types search locations */
-    typeRoots?: string[];
-    [option: string]: CompilerOptionsValue | undefined;
-}
+export type CompilerOptions = ChangePropertyTypes<ChangeStringIndexSignature<ts.CompilerOptions, CompilerOptionsValue>, {
+    jsx: JsxEmit | ts.JsxEmit;
+    module: ModuleKind | ts.ModuleKind;
+    moduleResolution: ModuleResolutionKind | ts.ModuleResolutionKind;
+    newLine: NewLineKind | ts.NewLineKind;
+    target: ScriptTarget | ts.ScriptTarget;
+}>;
 
 export const enum JsxEmit {
-    None = "None",
-    Preserve = "Preserve",
-    ReactNative = "ReactNative",
-    React = "React",
+    None = "none",
+    Preserve = "preserve",
+    ReactNative = "react-native",
+    React = "react",
+    ReactJSX = "react-jsx",
+    ReactJSXDev = "react-jsxdev",

```

Comment:
All of these compiler options enums were missing values, but this one also had a value `ReactNative = "ReactNative"` which has forever been broken, as far as I can tell. When inferred or external compiler options are sent over the wire, enum-typed properties are allowed to be, for example, `ts.JsxEmit | protocol.JsxEmit`; i.e. a number or a corresponding string value. However, the conversion method simply uses the same parser as command-line or tsconfig values, which means these string values need to match the ones in the maps defined in `commandLineParser.ts`, which are kebab-cased. Uppercase/lowercase letters don’t matter, but `"ReactNative"` would not have worked since it was missing the hyphen.

I changed all of these enum values to lowercase for consistency, but `react-native` was the only one that strictly matters. This is technically a type-level breaking change.